src/Magnum/Audio/Context.h

```diff
@@ -84,13 +131,215 @@ class MAGNUM_AUDIO_EXPORT Context {
+         * as it does most operations in compile time.
+         * @see @ref supportedExtensions(), @ref Extension::extensions(),
+         *      @ref MAGNUM_AUDIO_ASSERT_EXTENSION_SUPPORTED()
+         */
+        bool isExtensionSupported(const Extension& extension) const {
+            return _extensionStatus[extension._index];
+        }
+
     private:
         static Context* _current;
 
+        /* Create a context with given configuration. Returns `true` on success.
+         * @ref alcCreateContext(). */
+        bool tryCreateContext(const Configuration& config);
+
         ALCdevice* _device;
         ALCcontext* _context;
+
+        std::bitset<64> _extensionStatus;
+        std::vector<Extension> _supportedExtensions;
+};
+
+class MAGNUM_AUDIO_EXPORT Context::Configuration {
+    public:
+
+        /**
+         * @brief Enum for boolean values with a driver specific default
+         * value
+         */
+        enum class EnabledState: Byte {
+            Default = 0,
+            Enabled = 1,
+            Disabled = 2
+        };
+
+        /**
+         * @brief Constructor
+         */
+        explicit Configuration():
+            _frequency(44100),
+            _enableHrtf(),
+            _monoSources(-1),
+            _stereoSources(-1),
+            _refreshRate(-1)
+        {}
+
+        /** @brief Sampling rate in Hz */
+        Int frequency() const { return _frequency; }
+
+        /**
+         * @brief Set sampling rate (in Hz)
+         *
+         * Default is `44100`.
+         */
+        Configuration& setFrequency(Int freq) {
+            _frequency = freq;
+            return *this;
+        }
+
+        /** @brief Whether to use hrtfs */
+        EnabledState isHrtfEnabled() const { return _enableHrtf; }
+
+        /**
+         * @brief Set whether to use hrtfs
+         *
+         * Defaults to local OpenAL configuration or false.
+         */
+        Configuration& setHrtfEnabled(EnabledState hrtf) {
+            _enableHrtf = hrtf;
+            return *this;
+        }
+
+        /**
+         * @brief Hint for how mono sources to support
+         *
+         * Returns `-1`, if no hint was set.
+         */
+        Int monoSourcesCount() const { return _monoSources; }
+
+        /**
+         * @brief Set hint for how mono sources to support
+         *
+         * If not set, no hint will given to OpenAL.
+         */
+        Configuration& setMonoSourcesCount(Int sources) {
+            _monoSources = sources;
+            return *this;
+        }
+
+        /**
+         * @brief Hint for how stereo sources to support
+         *
+         * Returns `-1`, if no hint was set.
+         */
+        Int stereoSourcesCount() const { return _stereoSources; }
+
+        /**
+         * @brief Set hint for how stereo sources to support
+         *
+         * If not set, no hint will given to OpenAL.
+         */
+        Configuration& setStereoSourcesCount(Int sources) {
```

Comment:
Alright, I remembered that, but I wasn't sure whether this still holds when the implementation is in the header. Thank you for telling me again ;)